src/main/java/gift/repository/ProductRepositoryImpl.java

```diff
@@ -48,11 +48,36 @@ public Product save(Product product) {
     }
 
     @Override
-    public Product update(Product product) {
-        productMap.put(product.getId(), product); // 제품을 업데이트
-        return product; // 업데이트된 제품 반환
+    public Product updateNameById(Long productId, String name) {
+        if (productMap.containsKey(productId)) {
+            Product product = productMap.get(productId);
+            product.setName(name);
+            return product;
+        }
```

Comment:
이 부분이 updatePrice, updateImageUrl과 같이 중복되는 부분이 많은데, 어떻게 줄일 수 있을지 생각을 해 보셨을 것 같아요.
`함수형 인터페이스`라는 것이 존재하는데, 이를 활용하면 다음과 같이 중복을 제거 할 수 있을 것 같아요.
이런게 있다는 정도만 알고 넘어가도 좋아요.

```suggestion
private Product updateFieldById(Long productId, Consumer<Product> updater) {
    Product product = productMap.get(productId);
    if (product != null) {
        updater.accept(product);
    }
    return product;
}

public Product updateNameById(Long productId, String name) {
    return updateFieldById(productId, product -> product.setName(name));
}

@Override
public Product updatePriceById(Long productId, Long price) {
    return updateFieldById(productId, product -> product.setPrice(price));
}

@Override
public Product updateImageUrlById(Long productId, String imageUrl) {
    return updateFieldById(productId, product -> product.setImageUrl(imageUrl));
}
```

ref. https://inpa.tistory.com/entry/%E2%98%95-%ED%95%A8%EC%88%98%ED%98%95-%EC%9D%B8%ED%84%B0%ED%8E%98%EC%9D%B4%EC%8A%A4-API

Reply:
"PATCH와 같이 일부를 수정하기 위한 기능들도 만들어야 하지 않을까?" 라는 생각에서 각 필드에 대한 개별 구현을 만들었습니다. 
 PutMapping에 대한 구현에서는 Product로 받아 null checking후 기존 엔티티에 적용하였고 현재로서는 개별 구현은 사용되지 않습니다.
 updater로 권한을 위임한다면 바뀐 Product의 내용을 확인할 수 있어야 dao 단에서 개별 적용가능한데  그러기 위한 비용이 너무 커져서 어떻게 할 지 의문입니다.